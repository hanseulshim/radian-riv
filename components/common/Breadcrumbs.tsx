import React, { useState } from 'react'
import Link from 'next/link'
import Sidebar from 'components/Sidebar'
import { Route } from 'utils/routes'

type Props = {
  label: string
  current: string
  parentPath: string
  parents?: {
    path: string
    name: string
  }[]
  routes: Route[]
}
export default function Breadcrumbs({
  current,
  label,
  parentPath,
  parents = [],
  routes = []
}: Props) {
  const width = 310
  const [xPosition, setX] = useState<number>(-width)

  const toggleMenu = () => {
    if (xPosition < 0) {
      setX(0)
    } else {
      setX(-width)
    }
  }

  return (
    <>
      <div className="flex mb-4 space-x-2">
        <img
          id="sidebar"
          className="mr-4 cursor-pointer"
          alt="hamburger"
          src={`${process.env.baseUrl}/images/hamburger.svg`}
          onClick={toggleMenu}
        />
        <Link href={'/'}>
          <a className="underline text-patina">Home</a>
        </Link>
        {parents.map(({ name, path }) => (
          <React.Fragment key={name}>
            <span>/</span>
            <Link href={path}>
              <a className="underline text-patina">{name}</a>
            </Link>
          </React.Fragment>
        ))}
        <span>/</span>
        <span>{current}</span>
      </div>
      <Sidebar
        routes={routes}
        label={label}
        parentPath={parentPath}
        xPosition={xPosition}
        toggleMenu={toggleMenu}
      />
    </>
  )
}
