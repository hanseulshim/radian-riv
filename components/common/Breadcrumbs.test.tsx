import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Breadcrumbs from './Breadcrumbs'

describe('Breadcrumbs', () => {
  let component
  beforeAll(() => {
    const props = {
      current: 'Test',
      parents: [{ path: 'parent', name: 'parent' }],
      routes: [],
      label: 'county',
      parentPath: '/'
    }
    component = shallow(<Breadcrumbs {...props} />)
  })
  it('should have links to parent', () => {
    expect(component.find('a').at(0).text()).toEqual('Home')
    expect(component.find('a').at(1).text()).toEqual('parent')
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
