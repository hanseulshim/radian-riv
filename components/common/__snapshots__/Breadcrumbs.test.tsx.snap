// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should match snapshot 1`] = `
<Fragment>
  <div
    className="flex mb-4 space-x-2"
  >
    <img
      alt="hamburger"
      className="mr-4 cursor-pointer"
      id="sidebar"
      onClick={[Function]}
      src="undefined/images/hamburger.svg"
    />
    <Link
      href="/"
    >
      <a
        className="underline text-patina"
      >
        Home
      </a>
    </Link>
    <span>
      /
    </span>
    <Link
      href="parent"
    >
      <a
        className="underline text-patina"
      >
        parent
      </a>
    </Link>
    <span>
      /
    </span>
    <span>
      Test
    </span>
  </div>
  <SidebarNew
    label="county"
    parentPath="/"
    routes={Array []}
    toggleMenu={[Function]}
    xPosition={-310}
  />
</Fragment>
`;
